src/Microsoft.AspNet.Mvc/MvcServices.cs

```diff
@@ -89,6 +88,8 @@ public static IEnumerable<IServiceDescriptor> GetDefaultServices(IConfiguration
             yield return describe.Transient<IOutputFormattersProvider, DefaultOutputFormattersProvider>();
             yield return describe.Instance<JsonOutputFormatter>(new JsonOutputFormatter());
 
+            yield return describe.Transient<IModelBinderActivator, DefaultModelBinderActivator>();
```

Comment:
I know we always say 'transient by default' but services that have a cache of their own is enough to make this a non-default case. 

Registering a transient that:
1). DIs the service provider (which one?? it's a mystery)
2). caches data 

Seems really reckless.